File: leetcode/119/solution.py
"""
https://leetcode.com/problems/pascals-triangle-ii/
119. Pascal's Triangle II
"""
from typing import List


class Solution:
    def getRow(self, i: int) -> List[int]:
        if i == 0:
            return [1]

        arr = [1, 1]
        while i > 1:
            for j in range(len(arr) - 1):
                arr[j] += arr[j + 1]
            arr = [1] + arr[:-1] + [1]
            i -= 1

        return arr

File: leetcode/119/solution.rs
/**
 * https://leetcode.com/problems/pascals-triangle-ii/
 * 119. Pascal's Triangle II
 */

impl Solution {
    pub fn get_row(i: i32) -> Vec<i32> {
        let i = i as usize;
        let mut arr = vec![0; i + 1];
        arr[0] = 1;

        for j in 1..=i {
            for k in (1..=j).rev() {
                arr[k] += arr[k - 1];
            }
        }

        arr
    }
}
